File: UCF HSPT Documents/2013/Solutions/chairs.c

// 27th Annual UCF High School Programming Tournament
//
// Problem Name: Musical Chairs (chairs)
//
// Author:        Matt Fontaine
// Judge Data:    Marcos Arribas
// C Solution:    Nick Buelich
// Java Solution: Danny Wasserman
// Verifier:      Mike Galletti

#include <stdio.h>

main()
{
   // Open the input file
   FILE* fp = fopen("chairs.in", "r");

   // Read in the number of cases
   int cases = 0;
   fscanf(fp, "%d", &cases);

   // Loop over the cases
   int i = 0;
   for(i = 1; i <= cases; i++) {
      // Read in the number of children and chairs
      int a=0,b=0;
      fscanf(fp, "%d %d", &a, &b);  

      // Output how many children are eliminated during the round
      printf("Round #%d: %d children eliminated\n", i, a-b);
   }

   // Exit
   fclose(fp);
   return 0;
}

File: UCF HSPT Team/src/y2013/chairs.java

// 27th Annual UCF High School Programming Tournament
//
// Problem Name: Musical Chairs (chairs)
//
// Author:        Matt Fontaine
// Judge Data:    Marcos Arribas
// C Solution:    Nick Buelich
// Java Solution: Danny Wasserman
// Verifier:      Mike Galletti

import java.io.File;
import java.io.IOException;
import java.util.Scanner;


public class chairs {
  public static void main(String[] args) throws IOException {
    // Open file for reading
    Scanner in = new Scanner(new File("chairs.in"));

    // Get in the number of test cases
    int cases = in.nextInt();

    // Loop over the test cases
    for(int caseOn = 1; caseOn <= cases; caseOn++) {
      // Read in number of children and number of chairs
      int children = in.nextInt();
      int chairs = in.nextInt();

      // Since chairs <= children, can simply subtract.
      System.out.printf("Round #%d: %d children eliminated\n",
                        caseOn, children-chairs);
    }

    // Close
    in.close();
  }
}
